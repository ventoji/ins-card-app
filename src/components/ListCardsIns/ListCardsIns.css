.ins-cards-list {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    /*border: 1px solid black;*/
    background-color: rgba(206, 196, 196, 0.8);
    height: 89.8vh;
    border: none;
    overflow-y: scroll;
    scrollbar-width: none;
}
