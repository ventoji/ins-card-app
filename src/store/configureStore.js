import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import loggerMiddleware from './logger';
// import modalReducer from './modalReducer';
import cardReducer from './cardReducer';

//    modalOpen: modalReducer,
const rootReducer = combineReducers({
    cardList: cardReducer
});

const devTools =  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();

const middlewares = [];

if (process.env.NODE_ENV === 'development') {
    middlewares.push(loggerMiddleware);
  }
const middlewareEnhancer = applyMiddleware(...middlewares);
const enhancers = [middlewareEnhancer];

if (process.env.NODE_ENV === 'development') {
    enhancers.push(devTools);
  }


const composeEnhancers = compose(...enhancers);

/**
 * store for the card list app
 */
export default () => {
    const store = createStore(
        rootReducer,
        undefined,
        composeEnhancers
    );

      return store;
}