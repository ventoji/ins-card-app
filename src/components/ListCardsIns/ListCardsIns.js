import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import CardIns from "../CardIns";
import "./ListCardsIns.css";
import SortCardOptions from "../SortCardOptions";
import { sortByCards } from "./selectors/selector";

const INTIIAL_FILTER = {
  titleUser: "",
  title: false,
  createdAt: false,
  updatedAt: false,
};
/** List of cards Component */
export const ListCardsIns = ({ 
    handleModal,
    cardListStore 
  }) => {
  const [filteredOptions, setFilteredOptions] = useState(INTIIAL_FILTER);


  const handleEditItem = (id) => {
    //  console.log('PARENT', id);
    handleModal(id);
  };

  useEffect(() => {
    setFilteredOptions(filteredOptions);
  }, [filteredOptions,cardListStore]);

  const handleFilter = (e) => {
    //  console.log('DATA',e)
    setFilteredOptions(e);
  };

  return (
    <>
      <SortCardOptions onSubmit={handleFilter} />
      
      <div className="ins-cards-list">
        {cardListStore && cardListStore.length > 0 ? (
          sortByCards(cardListStore, filteredOptions).map((card) => 
             <CardIns key={card.id} {...card} handleModal={handleEditItem} />
          )
        ) : (
          <span>Not cards to display</span>
        )}
      </div>
    </>
  );
};

ListCardsIns.defaultProps = {
  cardsListStore: [],
  handleModal: () => {},
};

ListCardsIns.propTypes = {
  /**
   *  Arrays of cards.
   */
  cardsListStore: PropTypes.array,

  /**
   *  Handle modal window in children
   */
  handleModal: PropTypes.func,
};

const mapStatetoProps = (state) => ({
    modalOpen: state.modalOpen,
    cardListStore: state.cardList
});

export default connect(mapStatetoProps)(ListCardsIns);

