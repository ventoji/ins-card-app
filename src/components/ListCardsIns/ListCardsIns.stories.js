import React from "react";

import ListCardsIns from './ListCardsIns';

export default {
    title: "Components/ListCardsIns",
    component: ListCardsIns
  };

const CARD_DETAILS = {
    title: 'Moto GP',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus e',
    imageUrl: 'https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg'
};

const cardsList = [
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS,
    CARD_DETAILS
];

const Template = (args) => 
<div className="ins-row-stories-list-cards">
    <ListCardsIns {...args} />
</div>
;
  
export const CardsList = Template.bind({});
CardsList.args = {
    cardsList: cardsList
  };