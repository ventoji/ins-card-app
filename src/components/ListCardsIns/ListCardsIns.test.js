import React from "react";
import { render } from "@testing-library/react";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import {ListCardsIns} from "./ListCardsIns";
import TestRenderer from "react-test-renderer";

const CARD_DETAILS = {
  id: "sdhai123",
  title: "Moto GP",
  description: "Some description for the card, very nice",
  imageUrl:
    "https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg",
};

const CARD_DETAILS1 = {
  id: "sdhai323",
  title: "Moto GP",
  description: "Some description for the card, very nice",
  imageUrl:
    "https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg",
};

const CARD_DETAILS2 = {
  id: "sdhai193",
  title: "Moto GP",
  description: "Some description for the card, very nice",
  imageUrl:
    "https://images.freeimages.com/images/large-previews/adf/sun-burst-1478549.jpg",
};

const cardsList = [CARD_DETAILS, CARD_DETAILS1, CARD_DETAILS2];
const mockStore = configureStore([]);

describe("ListCardIns component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      state: {
        countries: false,
        cardListStore:cardsList
      },
    });

  });

  it("should be defined", () => {
    const { container } = render(
    <Provider store={store}>
      <ListCardsIns />
    </Provider>);
    expect(container).toBeInTheDocument();
  //  expect(container.querySelectorAll(".ins-card")).toHaveLength(3);
  });

  it('should render properly', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
      <ListCardsIns />
    </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
  })

  it("should render a message when no cards to display", () => {
    const { getByText } = render(<ListCardsIns />);
    let element = getByText("Not cards to display");
    expect(element.nodeName).toBe("SPAN");
  });
});
