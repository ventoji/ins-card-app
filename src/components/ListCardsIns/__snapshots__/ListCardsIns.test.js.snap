// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCardIns component should render properly 1`] = `
Array [
  <div
    className="ins-sort-card-options"
  >
    <form
      className="ins-form-sort"
      id="ins-form-sort"
      onSubmit={[Function]}
    >
      <div
        className="ins-form__group undefined"
        style={
          Object {
            "marginBottom": "1.5rem",
          }
        }
      >
        <label
          className="ins-form__label"
          htmlFor="titleUser"
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          Filter by title
           
        </label>
        <input
          className="ins-form__input"
          id="titleUser"
          name="titleUser"
          onChange={[Function]}
          placeholder="Filter by title card here"
          style={Object {}}
          type="text"
        />
      </div>
      <div
        className="ins-form__group undefined"
        style={
          Object {
            "marginBottom": "1.5rem",
          }
        }
      >
        <label
          className="ins-form__label"
          htmlFor="title"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Sort by title
           
        </label>
        <input
          className="ins-form__input"
          id="title"
          name="title"
          onChange={[Function]}
          placeholder="Title"
          style={Object {}}
          type="checkbox"
        />
      </div>
      <div
        className="ins-form__group undefined"
        style={
          Object {
            "marginBottom": "1.5rem",
          }
        }
      >
        <label
          className="ins-form__label"
          htmlFor="createdAt"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Filter by date
           
        </label>
        <input
          className="ins-form__input"
          id="createdAt"
          name="createdAt"
          onChange={[Function]}
          placeholder="Title"
          style={Object {}}
          type="checkbox"
        />
      </div>
      <div
        className="ins-form__group"
      >
        <label
          className="ins-form__label"
          htmlFor="sortBy"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          sort by
           
        </label>
        <select
          className="ins-select"
          id="sortBy"
          name="sortBy"
          onChange={[Function]}
          value="asc"
        >
          <option
            disabled={true}
            value={true}
          >
             
            -- select an option --
             
          </option>
          <option
            value="asc"
          >
            ascendent
          </option>
          <option
            value="desc"
          >
            descendent
          </option>
        </select>
      </div>
      <div
        className="ins-form__group"
      >
        <button
          className="ins-button ins-button--small ins-button--primary"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1rem",
              "marginTop": "2.4rem",
              "position": "relative",
              "top": "-2.6rem",
            }
          }
          type="submit"
        >
          find
        </button>
      </div>
    </form>
  </div>,
  <div
    className="ins-cards-list"
  >
    <span>
      Not cards to display
    </span>
  </div>,
]
`;
